src/core/thread/mle.cpp

```diff
@@ -2794,11 +2795,14 @@ void Mle::HandleUdpReceive(Message &aMessage, const Ip6::MessageInfo &aMessageIn
 #ifndef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
     // Intentionally ignore security check failures for broadcast MLE messages because it can be common to receive
     // such kind of messages from adjacent Thread networks.
-    VerifyOrExit(memcmp(messageTag, tag, sizeof(tag)) == 0,
-                 error = ((aMessageInfo.GetSockAddr().IsMulticast() &&
-                           aMessageInfo.GetThreadLinkInfo()->GetPanId() == Mac::kPanIdBroadcast)
-                              ? kErrorNone
-                              : kErrorSecurity));
+    VerifyOrExit(memcmp(messageTag, tag, sizeof(tag)) == 0, {
+        skipLoggingError = (aMessageInfo.GetSockAddr().IsMulticast() &&
+                            aMessageInfo.GetThreadLinkInfo()->GetPanId() == Mac::kPanIdBroadcast);
+        if (!skipLoggingError)
+        {
+            error = kErrorSecurity;
+        }
```

Comment:
Thanks for changing this. 👍 

I suggest keeping the `error` as is (since it is really a security error). We just set `skipLoggingError` so we don't log it. One other suggestion is whether to change this whole block to `if()` instead of `VerifyOrExit()`. I feel makes easier to read: 
```c++
    if (memcmp(messageTag, tag, sizeof(tag)) != 0)
    {
        // We skip logging security check failures for broadcast MLE
        // messages since it can be common to receive such messages 
        // from adjacent Thread networks.

        skipLoggingError = (aMessageInfo.GetSockAddr().IsMulticast() &&
                            aMessageInfo.GetThreadLinkInfo()->GetPanId() == Mac::kPanIdBroadcast);
        ExitNow(error = kErrorSecurity);
    }
```

Reply:
If we are not going to use it, is it better not set it?

Reply:
I think we may end up using the `error` in future (behaving differently on certain errors - I have in my mind some recent discussions and JIRAs on Thread TC calls related to MLE security failures and how to handle them). If we change the `error` here, later on when potentially adding new code or behaviors, we may not immediately remember or notice that the error was not set here in this case.

Also setting the error when the tag check fails seems to be fewer lines of code, otherwise, we need to add extra `if` check to explicitly not set it (`if (!skipLoggingError) { error = kErrorSecurity; }`).

Reply:
I was afraid it would generate set but unused warnings in static analysis check. However I just verified it doesn't. :)